timesketch/tsctl.py

```diff
@@ -1373,3 +1416,574 @@ def list_config():
+    "--output-format",
+    type=click.Choice(["csv", "jsonl"], case_sensitive=False),
+    default="csv",
+    help="Format for event data export (csv or jsonl). Default: csv",
+)
+@click.option(
+    "--filename",
+    required=False,
+    help=(
+        "Filename for the output zip archive. "
+        f"(Default: {DEFAULT_EXPORT_ARCHIVE_FILENAME_TEMPLATE})"
+    ),
+)
+@click.option(
+    "--all-fields",
+    is_flag=True,
+    default=True,
+    help="Export all event fields instead of the default set.",
+)
+def export_sketch(sketch_id: int, output_format: str, filename: str, all_fields: bool):
+    """Exports a Timesketch sketch to a zip archive.
+
+    The archive includes sketch metadata (as 'metadata.json') and all associated
+    events, formatted as specified (CSV or JSONL). By default, only a predefined
+    set of common fields are exported. Use the --all-fields flag to export all
+    available fields for each event. Progress messages are printed to the console
+    during the export process.
+
+    **WARNING:** Re-importing this archive into Timesketch is not natively
+    supported. This export is primarily for data archival, external analysis,
+    or manual migration.
+
+    Note: When running this command within a container (e.g., Docker),
+    the output zip file is written inside the container's filesystem.
+    Ensure you write to a mounted volume or copy the file out of the
+    container afterwards.
+    """
+    sketch = sketch = Sketch.get_by_id(sketch_id)
+    if not sketch:
+        print(f"ERROR: Sketch with ID {sketch_id} not found.")
+        return
+
+    if not filename:
+        filename = DEFAULT_EXPORT_ARCHIVE_FILENAME_TEMPLATE.format(
+            sketch_id=sketch_id, output_format=output_format
+        )
+
+    if not filename.lower().endswith(".zip"):
+        filename += ".zip"
+
+    print(f'Exporting sketch [{sketch_id}] "{sketch.name}" to {filename}...')
+
+    # --- Add prominent warning to console output ---
+    click.echo(
+        click.style(
+            "\nWARNING: There is currently no native method to re-import "
+            "this exported archive back into Timesketch.\n",
+            fg="yellow",
+            bold=True,
+        ),
+        err=True,  # Print to stderr to make it more noticeable
+    )
+    # --- End warning ---
+
+    try:
+        # 1. Gather Metadata
+        metadata = _get_sketch_metadata(sketch)
+
+        # 2. Fetch and Prepare Event Data
+        # Get datastore instance
+        datastore = OpenSearchDataStore(
+            host=current_app.config["OPENSEARCH_HOST"],
+            port=current_app.config["OPENSEARCH_PORT"],
+        )
+
+        if all_fields:
+            print("  Exporting all event fields.")
+            return_fields_to_fetch = None  # Pass None to get all fields
+        else:
+            print(f"  Exporting default fields: {', '.join(DEFAULT_SOURCE_FIELDS)}")
+            return_fields_to_fetch = DEFAULT_SOURCE_FIELDS  # Use default fields
```

Comment:
I used `tsctl export-sketch 1 --all-fields` but I still got the message `Exporting default fields: datetime, timestamp, timestamp_desc, _index, __ts_timeline_id, message, comment, timesketch_label, tag, similarity_score, human_readable, __ts_emojis`. 

But when I use `tsctl export-sketch 1` I get `Exporting all event fields.`.

Somehow this is the wrong way round?

Reply:
ah yes, let me change that, I made the default to export all...